### ISSUE TITLE
Rule proposal: `no-delete-operator`

### ISSUE THREAD

**Issue by jendrikw** [sindresorhus/eslint-plugin-unicorn#2800]
### Description

Lints against every use of the delete operator.

### Examples

```js
// ❌
let a = {x:1, y:2}
delete a.x;

// ✅
let a = {x:1, y:2}
let {x, ...a2} = a;
```

### Proposed rule name

no-delete-operator

### Additional Info

There is https://eslint.org/docs/latest/rules/no-delete-var, which delete with `delete variableName` and nothing else. For arrays, there's https://typescript-eslint.io/rules/no-array-delete/.

I want all usage of the delete keyword banned.

**Comment by fregante** [comment]
> Can you explain why `delete object.property` is bad?

Likely they prefer immutability. So not bad, just a preference.

**Comment by sindresorhus** [comment]
Passing on this as it's too niche and already possible.
